### PR TITLE
fix(ngrx-store): use ngrx store feature selectors in Stark Core modules. Use store-freeze and store-logger meta-reducers in Starter

### PR DESCRIPTION
**Pull request by christophercr**
Solves #94 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[X] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

All Stark modules defined the NGX-Store by calling `forRoot`. This causes an exception since `forRoot` should only be used once and only by the Application main module.

Besides that, it is not possible to debug the Application State composition in the Starter app.

Issue Number: #94 

## What is the new behavior?
All Stark modules define the store by calling `forFeature` and provide a [feature store selector](https://github.com/ngrx/platform/blob/master/docs/store/selectors.md#createfeatureselector) as described here: [NGRX-Store API](https://github.com/ngrx/platform/blob/master/docs/store/api.md).

To provide better tooling while defining and implementing the app NGRX store, the [ngrx-store-logger](https://github.com/btroncone/ngrx-store-logger) and the [ngrx-store-freeze](https://github.com/brandonroberts/ngrx-store-freeze) were added to the Starter app.

**IMPORTANT: These store meta-reducers are in fact a developer's choice and not Stark's.**

## Does this PR introduce a breaking change?
```
[X] Yes
[ ] No
```

### CODE REVIEW

File: packages/stark-core/src/logging/logging.module.ts [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-import { loggingReducer } from "./reducers/index";
+import { loggingReducer, StarkLoggingState } from "./reducers/index";
 import { StarkLoggingActions } from "./actions/index";
 import { StarkLoggingServiceImpl, starkLoggingServiceName } from "./services/index";
-import { StarkLoggingApplicationState } from "../common/store/starkCoreApplicationState";
 
-const reducers: ActionReducerMap<StarkLoggingApplicationState, StarkLoggingActions> = {
-	starkLogging: loggingReducer
+const reducers: ActionReducerMap<StarkLoggingState, StarkLoggingActions> = {
+	logging: loggingReducer
```
Comment (dsebastien) [inline packages/stark-core/src/logging/logging.module.ts]: Just curious, why the change of name?
Does this change the store structure? If so then I'd prefer to keep our prefixes.

Comment (christophercr) [reply packages/stark-core/src/logging/logging.module.ts]: Well, since we define now a "feature" store for every core module, then what we have in the Application Store is something like this:

```javascript
{
  MainAppState :{ }  // root state
  StarkLogging: {  // feature state => with prefix
    logging: ...   // without prefix otherwise it is too verbose having another "starkLogging" inside
  }
  StarkSession: {  // the same principle here
    session: ...
  }
}
```

File: packages/stark-core/src/logging/services/logging.service.ts [hunk 2]
```diff
@@ -77,12 +78,10 @@ export class StarkLoggingServiceImpl implements StarkLoggingService {
 			this.logUrl = this.backend.url + "/" + this.appConfig.loggingFlushResourceName;
 			this.generateNewCorrelationId();
 
-			this.store
-				.select<StarkLogging>((state: StarkLoggingApplicationState) => state.starkLogging)
-				.subscribe((starkLogging: StarkLogging) => {
-					this.starkLogging = starkLogging;
-					this.persistLogMessages();
-				});
+			this.store.pipe(select(selectStarkLogging)).subscribe((starkLogging: StarkLogging) => {
```
Comment (SuperITMan) [inline packages/stark-core/src/logging/services/logging.service.ts]: Just a suggestion, could we maybe use starkLoggingSelector instead of selectStarkLogging ? :)
It's weird for me having select(select<>). I would prefer if we had something as select(<>Selector)...

Comment (christophercr) [reply packages/stark-core/src/logging/services/logging.service.ts]: Mmm I see, well it depends on which point of view you look at it, in my case I thought of this:

1. normally the selector functions return (select) something from the state and as such, the name of the function is imperative: "selectXXXX"
2. follow the guidelines defined in the NGRX-Store API, since these functions will be exposed, keeping in mind that now any developer can use our framework and we should try not to diverge too much (naming conventions, API, ...) from the rest of the ecosystem. Otherwise my fear is that the learning curve of our framework, which is already important, will deter the devs from adopting it
